src/devices_models/devices/hydro_generation.jl

```diff
@@ -526,24 +576,26 @@ function device_energy_limit_param_ub(
     return
 end
 
-function device_energy_limit_ub(
+function device_energy_budget_ub(
     psi_container::PSIContainer,
-    energy_limit_constraints::Vector{DeviceTimeSeriesConstraintInfo},
+    energy_budget_constraints::Vector{DeviceTimeSeriesConstraintInfo},
     cons_name::Symbol,
     var_name::Symbol,
 )
     time_steps = model_time_steps(psi_container)
     variable = get_variable(psi_container, var_name)
-    names = (get_name(x) for x in energy_limit_constraints)
+    names = (get_name(x) for x in energy_budget_constraints)
     constraint = add_cons_container!(psi_container, cons_name, names)
 
-    for constraint_info in energy_limit_constraints
+    for constraint_info in energy_budget_constraints
         name = get_name(constraint_info)
+        resolution = model_resolution(psi_container)
+        inv_dt = 1.0 / (Dates.value(Dates.Second(resolution)) / SECONDS_IN_HOUR)
         forecast = constraint_info.timeseries
-        multiplier = constraint_info.multiplier
+        multiplier = constraint_info.multiplier * inv_dt
         constraint[name] = JuMP.@constraint(
             psi_container.JuMPmodel,
-            sum([variable[name, t] for t in time_steps]) <= multiplier * sum(forecast)
+            sum([variable[name, t] for t in time_steps]) <= multiplier * forecast[end]
```

Comment:
I don't understand this change. What is significance of the end of the energy_budget forecast? Since we can't have mixed resolution forecasts, we had previously implemented it with the assumption that the energy budget was distributed evenly across the resolution of the forecast. e.g. if you have daily energy budget data, and an hourly `System` you would make an energy_budget forecast that had (daily energy budget)/24 in each hour. In this case, you would have to sum the RHS of the constraint. I think your case would require repeating the daily energy budget 24x since you are just using one value.

Reply:
It is up to us here. After discussing with @jd-lara, the idea was to modify the time series to be the same value the entire time series (i.e. each entry will be the budget), and the value that matters is the last one.

However, this idea of having the budget divided by 24 is also an option. It is up to us to decide how we want the data to be provided.
This also will appear as a similar constraint for the energy capacity in the ReservoirStorage, on which we want that the energy at the end of the time series to be greater than the proportion of the storage capacity. Again we may want to decide how we are going to provide the time series. As long as we stick with one we can decide use the similar approach for both. I was also thinking on generalizing the labels here, and instead of `hydro_budget` and `energy_bound, we can go with `storage_energy_limit` or something similar that has different interpretation depending on the model.

Reply:
I am not sure if adding many paramenters and their multipliers is the way to go with the budget. That formulation is pretty messy and it might even need time resolution conversions. We can discuss more about the implications of each.

Reply:
Using the last of repeated values means that whatever forecast is given for the budget has to be carefully calculated for the specific problem horizon. By dividing a budget into average values per time period, the op_problem can sum over whatever horizon it represents to get a valid energy budget.